File: backend/melodiket/src/main/java/com/ssafy/jdbc/melodiket/stage/entity/Stage.java
package com.ssafy.jdbc.melodiket.stage.entity;

import com.ssafy.jdbc.melodiket.concert.entity.ConcertEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "stage")
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Stage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private UUID uuid;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String address;

    @Column(nullable = false)
    private Boolean isStanding;

    private Long numOfRow;
    private Long numOfCol;

    @Column(nullable = false)
    private Long capacity;

    @OneToMany(mappedBy = "stage", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<StageAssignment> stageAssignments = new ArrayList<>();

    @OneToMany(mappedBy = "stage", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ConcertEntity> concerts = new ArrayList<>();

}

File: frontend/melodiket/src/types/stage.ts
import { UUID } from 'crypto';

export interface StageData {
  name: string;
  address: string;
  isStanding: boolean;
  capacity?: number;
  numOfRow?: number;
  numOfCol?: number;
}

export interface RegisterStandingStageRequest {
  name: string;
  address: string;
  capacity: number;
}

export interface RegisterSeatingStageRequest {
  name: string;
  address: string;
  numOfRow: number;
  numOfCol: number;
}

export interface RegisterStageResponse {
  uuid: UUID;
  name: string;
  address: string;
  isStanding: boolean;
  capacity: number;
  numOfRow: number;
  numOfCol: number;
}
